src/activity_item_handling.cpp

```diff
@@ -806,6 +808,29 @@ void activity_on_turn_move_loot( player_activity &, player &p )
                     }
 
                     if( g->m.free_volume( dest_loc ) > it->volume() ) {
+                        // get route to source tile, if we didn't get it previously
+                        if( !has_route ) {
+                            route = g->m.route( p.pos(), src_loc, p.get_pathfinding_settings(), p.get_path_avoid() );
```

Comment:
Is getting the pathfinding settings and path avoidance necessary every loop?

Reply:
If by every loop you mean once per each source tile that actually has loot to sort.

get_pathfinding_settings() just returns pointer
get_path_avoid should be quite quick unless you see zillion NPCs

Much more computing power is lost by nature of activity processing. You do lot of stuff that you throw away once player turn moves reaches zero or negative and you return the execution back to the game. And you do it again and again each time the activity is recalled. Yes, you could possibly cache some to save computing, but you risk that the world changes in the meanwhile and break stuff in your cache.

Since this is call on demand activity and not something that runs all the time, optimization to the ground is probably not necessary right now, could always be improved upon later.